src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -126,7 +127,9 @@ ImmutabilityContext context
 				return true;
 			}
 
-			var decl = field.GetDeclarationSyntax<VariableDeclaratorSyntax>();
+			var decl = field.DeclaringSyntaxReferences.Single()
+				.GetSyntax() as VariableDeclaratorSyntax;

```

Comment:
Just inlining (softer cast though, if the cast were to fail `NullReferenceException` is fine, but if so there is a chance we don't need the syntax anyway)

Reply:
`.Single()` throws if more than one.  Should not be more than one for fields.

Reply:
Yep, I'm happy with that. `GetDeclarationSyntax` would do the same thing and as far as we know its not possible (e.g. "partial fields")